linkerd/trace-context/src/propagation/b3.rs

```diff
@@ -0,0 +1,163 @@
+use crate::propagation::try_split_to;
+use crate::{Flags, Id};
+use bytes::Bytes;
+use http::header::HeaderValue;
+use linkerd_error::Error;
+use rand::thread_rng;
+
+use tracing::{trace, warn};
+
+use super::{decode_id_with_padding, get_header_str, Propagation, TraceContext, UnknownFieldId};
+
+static HTTP_TRACE_ID_HEADER: http::header::HeaderName =
+    http::header::HeaderName::from_static("x-b3-traceid");
+static HTTP_SPAN_ID_HEADER: http::header::HeaderName =
+    http::header::HeaderName::from_static("x-b3-spanid");
+static HTTP_SAMPLED_HEADER: http::header::HeaderName =
+    http::header::HeaderName::from_static("x-b3-sampled");
+
+static GRPC_TRACE_HEADER: http::header::HeaderName =
+    http::header::HeaderName::from_static("grpc-trace-bin");
```

Comment:
tiny stype nit, not important: i would probably import `http::header::HeaderName` to make these a little bit less wordy, especially because `HeaderValue` is already imported.